sdk/Sdk.Generators/Constants.cs

```diff
@@ -5,41 +5,50 @@ namespace Microsoft.Azure.Functions.Worker.Sdk.Generators
 {
     internal static class Constants
     {
-        // MSBuild Props
-        internal const string EnableSourceGenProp = "build_property.FunctionsMetadataSourceGen_Enabled";
+        public static class BuildProperties
+        {
+            internal const string EnableSourceGenProp = "build_property.FunctionsMetadataSourceGen_Enabled";
+        }
 
-        // Generated File Names
-        internal const string GeneratedFunctionMetadataFileName = "GeneratedFunctionMetadataProvider.g.cs";
+        public static class FileNames
+        {
+            internal const string GeneratedFunctionMetadataFileName = "GeneratedFunctionMetadataProvider.g.cs";
```

Comment:
With the subclass, you can drop some of the suffixes now as the subclass covers that:

```suggestion
            internal const string GeneratedFunctionMetadata = "GeneratedFunctionMetadataProvider.g.cs";
```

This will still be clear in usage as we would write:

``` CSharp
Constants.FileNames.GeneratedFunctionMetadata;
```